dotnet/src/SemanticKernel.Abstractions/Contents/ClassificationContent.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Represents result of classification of content that is classified by an AI model.
+/// </summary>
+public class ClassificationContent : KernelContent
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ClassificationContent"/> class.
+    /// </summary>
+    /// <param name="innerContent">The inner content representation</param>
+    /// <param name="classifiedContent">Content which has been classified</param>
+    /// <param name="result">The results of content classification by an AI model.</param>
+    /// <param name="modelId">The model ID used to generate the content</param>
+    /// <param name="metadata">Metadata associated with the content</param>
+    public ClassificationContent(
+        object? innerContent,
+        string classifiedContent,
+        IReadOnlyDictionary<string, object?> result,
+        string? modelId = null,
+        IReadOnlyDictionary<string, object?>? metadata = null)
+        : base(innerContent, modelId, metadata)
+    {
+        this.ClassifiedContent = classifiedContent;
+        this.Result = result;
+    }
+
+    /// <summary>
+    /// Represents the classified content.
+    /// </summary>
+    public string ClassifiedContent { get; }
```

Comment:
I think this abstraction is still too abstract (Results Dictionary of `objects?`)

Is not very clear what to expect of the value in the `ClassifiedContent`, is that the `Prompt` sent or ?

This might need more investigation and probably an ADR where we can have a better traversable content for classification models.

Looking at `HuggingFace` inference API for `Text-Classification` we have this result:

Hugging Face
```
[
    [
        {
            "label": "disappointment",
            "score": 0.5044490694999695
        },
        {
            "label": "sadness",
            "score": 0.3469429612159729
        },
        ...
    ]
]
```

Open AI
```
{
    "id": "modr-XXXXX",
    "model": "text-moderation-007",
    "results": [
        {
            "flagged": true,
            "categories": {
                "sexual": false,
                "hate": false,
                "harassment": false,
                "self-harm": false,
                "sexual/minors": false,
                "hate/threatening": false,
                "violence/graphic": false,
                "self-harm/intent": false,
                "self-harm/instructions": false,
                "harassment/threatening": true,
                "violence": true
            },
            "category_scores": {
                "sexual": 1.2282071e-6,
                "hate": 0.010696256,
                "harassment": 0.29842457,
                "self-harm": 1.5236925e-8,
                "sexual/minors": 5.7246268e-8,
                "hate/threatening": 0.0060676364,
                "violence/graphic": 4.435014e-6,
                "self-harm/intent": 8.098441e-10,
                "self-harm/instructions": 2.8498655e-11,
                "harassment/threatening": 0.63055265,
                "violence": 0.99011886
            }
        }
    ]
}
```

Reply:
I am also not convinced with this. But I don't see good way to abstract classification content because results from models are too different. So maybe we should remove abstraction and leave only implementation for classification service?